src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -302,9 +317,30 @@ public function configureOptions(OptionsResolver $resolver)
             return $choiceTranslationDomain;
         };
 
+        $multipleNormalizer = function (Options $options, $multiple) {
+            if (in_array($options['widget'], array('radio', 'checkbox'))) {
+                return 'checkbox' == $options['widget'];
+            }
+
+            return $multiple;
+        };
+
+        $expandedNomalizer = function (Options $options, $expanded) {
+            if (null !== $expanded) {
+                @trigger_error('The "expanded" option is deprecated since version 3.1 and will be removed in 4.0. You should use "widget" instead.', E_USER_DEPRECATED);
+            } else {
+                $expanded = false;
+            }
+
+            return in_array($options['widget'], array('radio', 'checkbox')) ?: $expanded;
+        };
+
```

Comment:
there is no need to handle `expanded` value when it is deprecated by this PR... this is actually the meaning of this normalizer.

you could write :

```php
$expandedNomalizer = function (Options $options, $expanded) {
    if (null !== $expanded) {
        @trigger_error('The "expanded" option is deprecated since version 3.1 for internal use only. You should use "widget" option with "radio" or "checkbox" instead.', E_USER_DEPRECATED);
     }

    return 'select' === $options['widget'] ? false : (in_array($options['widget'], array('radio', 'checkbox')) ?: $expanded);
};

only if `text` and `hidden` were indeed been handled differently when `expanded` is true and if it was not deprecated. Or neither options seems to be implemented properly in this PR.